### PR TITLE
Enable NormalizeL2Fusion and LeakyReluFusion inside MOC

### PR DESCRIPTION
**Pull request by GlebKazantaev**
### Description
* Refactored NormlaizeL2 fusions to use single pattern
* Enabled NormalizeL2Fusion and LeakyReluFusion inside MO transformation pipeline
Ticket: XXX-62414

### Validation:
* Functional validation shows no regressions
* IR comparison validation shows no difference in IRs which is expected

### nGraph Transformations Transition Status
<details>
  <summary>Click to expand!</summary>

|Final MOC Pipeline|Enabled|
|---|:---:|
|InitNodeInfo|:white_check_mark:|
|ConstantFolding|:white_check_mark: |
|SimplifyShapeOfSubGraph|:white_check_mark:|
|RemoveFilteringBoxesBySize|:white_check_mark:|
|ConvertQuantizeDequantize|:white_check_mark:|
| | |
|TransposeFQReduction|:white_check_mark:|
|TransposeReduction|:white_check_mark:|
|TransposeFuse|:white_check_mark:|
|SplitSqueezeConcatFusion|:white_check_mark:|
| | |
|EliminateUnsqueezeGather|:white_check_mark:|
|NopElimination|:white_check_mark: |
| | |
|ConvertScatterElementsToScatter|:white_check_mark:|
|BroadcastElementwiseFusion|:white_check_mark:|
|SoftPlusFusion|:white_check_mark:|
|SoftPlusToMishFusion|:white_check_mark:|
|SwishFusion|:white_check_mark:|
|HSwishFusion|:white_check_mark:|
|HSigmoidFusion|:white_check_mark:|
|**NormalizeL2Fusion**|:tada: |
|ClampFusion|:white_check_mark:|
|PadFusion|:white_check_mark:|
|SoftmaxFusion| |
|MVNFusion|:white_check_mark:| 
|DilatedConvolutionConverter|:white_check_mark:|
|GeluFusion|:white_check_mark:|
|**LeakyReluFusion**|:tada:|
| | |
|BinarizeWeights|:white_check_mark:|
|ConvToBinaryConv|:white_check_mark:|
| | |
|BatchNormDecomposition|:white_check_mark: |
|ConvertDivide||
|ConvertSubtract|| 
| | |
|LinOpSequenceFusion|:white_check_mark: |
| | |
|ConvolutionMultiplyFusion|:white_check_mark: |
|GroupConvolutionMultiplyFusion|:white_check_mark: |
|ConvolutionBackpropDataMultiplyFusion|:white_check_mark: |
|GroupConvolutionBackpropDataMultiplyFusion|:white_check_mark: |

_Note: transformations marked as **bold** and with :tada: were enabled in current PR_
</details>

### CODE REVIEW

**Review by lazarevevgeny — APPROVED** [review]
Looks amazing. The only comment is that may be you can remove the NormalizeL2 fusion transformation in the MO in this PR so we test the whole pipeline?

**Comment by GlebKazantaev** [comment]
> Looks amazing. The only comment is that may be you can remove the NormalizeL2 and LeakyRelu fusion transformation in the MO in this PR so we test the whole pipeline?

If we disable this passes in MO there is a risk that some MO fusions can break this patterns so finally this nGraph transformations won't be applied.
